### PR TITLE
[NP=8651] Re-send webhook on 4xx and 5xx

### PR DESCRIPTION
**Pull request by jinnnn9**
https://nanopay.atlassian.net/browse/NP-8651

### CODE REVIEW

File: src/foam/nanos/dig/HTTPDigestSink.java [hunk 1]
```diff
@@ -79,49 +78,46 @@ public void put(Object obj, Detachable sub) {
     FObject fobj = (FObject) obj;
     Object id = fobj.getProperty("id");
     String className = fobj.getClass().getSimpleName();
+    int responseCode = -1;
+    int count = 3;
```
Comment (adamfox-nanopay) [inline src/foam/nanos/dig/HTTPDigestSink.java]: Can we call this something more meaningful like retryCount

File: src/foam/nanos/dig/HTTPDigestSink.java [hunk 2]
```diff
@@ -79,49 +78,46 @@ public void put(Object obj, Detachable sub) {
...
+        throw new RuntimeException(e);
+      } catch (Throwable t) {
+        String name = "HTTP DIGEST UNEXPECTED FAILURE";
         String note = "[" + className + ", " + id + ", " + new Date() + "]";
-        createAlarm(name, note, LogLevel.WARN);
-      }
+        createAlarm(name, note, LogLevel.ERROR);
 
-      throw new RuntimeException(this.getClass().getSimpleName() + "[" + className + ", " + id + ", " + responseCode + "]");
+        throw new RuntimeException(t);
```
Comment (adamfox-nanopay) [inline src/foam/nanos/dig/HTTPDigestSink.java:111]: Whats the benefit of throwing RuntimeException here?

Comment (jinnnn9) [reply src/foam/nanos/dig/HTTPDigestSink.java:111]: sendRequest might fail for a reason other than socket error .. this catches that here

Comment (jinnnn9) [reply src/foam/nanos/dig/HTTPDigestSink.java:111]: and I believe all the exceptions thrown in the put are handled in the calling method

**Review by adamfox-nanopay — APPROVED** [review]

**Review by jlhughes — APPROVED** [review]

**Pull request merged**